job/server/src/main/java/alluxio/master/job/plan/PlanCoordinator.java

```diff
@@ -267,13 +269,28 @@ private synchronized void updateStatus() {
     int completed = 0;
     List<TaskInfo> taskInfoList = mPlanInfo.getTaskInfoList();
     for (TaskInfo info : taskInfoList) {
-      switch (info.getStatus()) {
+      JobConfig config = mPlanInfo.getJobConfig();
+      Preconditions.checkNotNull(config);
+      Status status = info.getStatus();
+      FileSystem fileSystem = mJobServerContext.getFileSystem();
+
+      switch (status) {
         case FAILED:
           setJobAsFailed(info.getErrorType(), "Task execution failed: " + info.getErrorMessage());
+          if (DistributedCmdMetrics.isBatchConfig(config)) {
```

Comment:
would prefer to deal with batch/non-batch situations in DistributedCmdMetrics class and keep here clean.

Reply:
done